prow/cmd/generic-autobumper/main.go

```diff
@@ -19,47 +19,467 @@ package main
 	var config string
 	var labelsOverride []string
+	var skipPullRequest bool
 
 	flag.StringVar(&config, "config", "", "The path to the config file for the autobumber.")
 	flag.StringSliceVar(&labelsOverride, "labels-override", nil, "Override labels to be added to PR.")
+	flag.BoolVar(&skipPullRequest, "skip-pullrequest", false, "")
 	flag.Parse()
 
-	var o bumper.Options
+	var o Options
+	var pro bumper.Options
 	data, err := ioutil.ReadFile(config)
 	if err != nil {
-		return nil, fmt.Errorf("Failed to read in config file, %s", config)
+		return nil, nil, fmt.Errorf("Failed to read in config file, %s", config)
+	}
+
+	err = yaml.Unmarshal(data, &o)
+	if err != nil {
+		return nil, nil, fmt.Errorf("Failed to parse yaml file, %s", err)
 	}
 
-	if err := yaml.UnmarshalStrict(data, &o); err != nil {
-		return nil, fmt.Errorf("Failed to parse yaml file, %s", err)
+	if err := yaml.Unmarshal(data, &pro); err != nil {
+		return nil, nil, fmt.Errorf("Failed to parse yaml file, %s", err)
 	}
 
 	if labelsOverride != nil {
-		o.Labels = labelsOverride
+		pro.Labels = labelsOverride
 	}
-	return &o, nil
+	pro.SkipPullRequest = skipPullRequest
+	return &o, &pro, nil
+}
+
+func validateOptions(o *Options) error {
+	if len(o.Prefixes) == 0 {
+		return fmt.Errorf("Must have at least one Prefix specified")
+	}
+	if len(o.IncludedConfigPaths) == 0 {
+		return fmt.Errorf("includedConfigPaths is mandatory")
+	}
+	if o.TargetVersion != latestVersion && o.TargetVersion != upstreamVersion &&
+		o.TargetVersion != upstreamStagingVersion && !tagRegexp.MatchString(o.TargetVersion) {
+		logrus.Warnf("Warning: targetVersion is not one of %v so it might not work properly.",
+			[]string{latestVersion, upstreamVersion, upstreamStagingVersion, tagVersion})
+	}
+	if o.TargetVersion == upstreamVersion {
+		for _, prefix := range o.Prefixes {
+			if prefix.RefConfigFile == "" {
+				return fmt.Errorf("targetVersion can't be %q without refConfigFile for each prefix. %q is missing one", upstreamVersion, prefix.Name)
+			}
+		}
+	}
+	if o.TargetVersion == upstreamStagingVersion {
+		for _, prefix := range o.Prefixes {
+			if prefix.StagingRefConfigFile == "" {
+				return fmt.Errorf("targetVersion can't be %q without stagingRefConfigFile for each prefix. %q is missing one", upstreamStagingVersion, prefix.Name)
+			}
+		}
+	}
+	if (o.TargetVersion == upstreamVersion || o.TargetVersion == upstreamStagingVersion) && o.UpstreamURLBase == "" {
+		o.UpstreamURLBase = defaultUpstreamURLBase
+		logrus.Warnf("targetVersion can't be 'upstream' or 'upstreamStaging` without upstreamURLBase set. Default upstreamURLBase is %q", defaultUpstreamURLBase)
+	}
+
+	return nil
+}
+
+// UpdateReferences update the references of prow-images and/or boskos-images and/or testimages
+// in the files in any of "subfolders" of the includeConfigPaths but not in excludeConfigPaths
+// if the file is a yaml file (*.yaml) or extraFiles[file]=true
+func UpdateReferences(o *Options) (map[string]string, error) {
```

Comment:
What package are you exporting UpdateReferences to?

If this is only used by the main package, please do not export this function unless another package needs it or you are trying to satisfy an interface (applies to all exported functions in the main package)

Reply:
renamed and made it private